filebeat/inputsource/udp/server.go

```diff
@@ -62,12 +60,11 @@ func (u *Server) createConn() (net.PacketConn, error) {
 	if err != nil {
 		return nil, err
 	}
-	socketSize := int(u.config.ReadBuffer) * humanize.KiByte
-	if socketSize != 0 {
-		if err := listener.SetReadBuffer(int(u.config.ReadBuffer)); err != nil {
-			return nil, err
-		}
+
+	if err := listener.SetReadBuffer(int(u.config.ReadBuffer)); err != nil {
```

Comment:
Is it truly safe to remove the check for 0?  The config only validates that it is positive:

https://github.com/elastic/go-ucfg/blob/587fe67620f33b205484ef84b547d1ad30531035/validator.go#L249-L276

Reply:
Good question! Eventually, it's passed to a OS specific implementation, for example this is for POSIX https://cs.opensource.google/go/go/+/master:src/net/sockopt_posix.go;l=75-79?q=setReadBuffer&ss=go%2Fgo

We cannot guarantee that every OS would return us a proper error when the buffer size is 0. I think we should put the zero check back here.

Reply:
Makes sense!! Let me add the condition back.